### PR TITLE
CS-407 Add `get  /annotations` API service to python SDK

### PR DESCRIPTION
**Pull request by Nutto55**

### CODE REVIEW

File: package-rfcx/rfcx/client.py [hunk 1]
```diff
@@ -264,3 +264,25 @@ def streams(self, keyword=None, limit=1000, offset=0):
...
         return api_rfcx.streams(self.credentials.id_token, keyword, limit, offset)
+
+    def annotations(self, start=None, end=None, classifications=None, stream_id=None, limit=1000, offset=0):
+        """Retrieve a list of annotations
+
+        Args:
+            start: Minimum timestamp of the audio. If None then defaults to exactly 30 days ago.
+            end: Maximum timestamp of the audio. If None then defaults to now.
+            classifications: (optional, default= None) List of classification names.
+            stream_id: (optional,default= None) Limit results to a given stream.
```
Comment (antonyharfield) [inline package-rfcx/rfcx/client.py:275]: Let's change `stream_id` to be `stream` for any public functions exposed by the SDK.

File: package-rfcx/rfcx/_api_rfcx.py [hunk 2]
```diff
@@ -14,6 +14,12 @@ def streamSegments(token, stream_id, start, end, limit, offset):
...
 
+def annotations(token, start, end, classifications=None, stream_id=None, limit=1000, offset=0):
+    data = {'start': start, 'end': end, 'classifications[]': classifications, 'stream_id': stream_id, 'limit': limit, 'offset': offset}
+    path = '/annotations'
+    url = '{}{}?{}'.format(host, path, urllib.parse.urlencode(data, True))
+    return _request(url, token=token)
+
 def tags(token, type, labels, start, end, sites, limit):
     data = {'type': type, 'values[]': labels, 'starting_after_local': start, 'starting_before_local': end, 'sites[]': sites, 'limit': limit}
     path = '/v2/tags'
```
Comment (antonyharfield) [inline package-rfcx/rfcx/_api_rfcx.py:25]: This one needs to be deleted.

File: package-rfcx/rfcx/_api_rfcx.py [hunk 3]
```diff
@@ -14,6 +14,12 @@ def streamSegments(token, stream_id, start, end, limit, offset):
     url = '{}{}?{}'.format(host, path, urllib.parse.urlencode(data, True))
     return _request(url, token=token)
 
+def annotations(token, start, end, classifications=None, stream_id=None, limit=1000, offset=0):
+    data = {'start': start, 'end': end, 'classifications[]': classifications, 'stream_id': stream_id, 'limit': limit, 'offset': offset}
```
Comment (antonyharfield) [inline package-rfcx/rfcx/_api_rfcx.py:18]: Stas added new parameters to this API last week.

Comment (antonyharfield) [reply package-rfcx/rfcx/_api_rfcx.py:18]: Should be `is_positive` and `is_something_else_that_i_cant_remember`

**Review by antonyharfield — CHANGES_REQUESTED** [review]

File: package-rfcx/rfcx/_api_rfcx.py (hunk 3 above)
Comment (Nutto55) [reply package-rfcx/rfcx/_api_rfcx.py]: From what have I check, it is increase an update on the `streams/:id/annotations` using `post` and it is for create an annotation belonging to a stream. I think it is not an endpoint to be on the SDK.

Comment (antonyharfield) [reply package-rfcx/rfcx/_api_rfcx.py]: Ok - we can talk about this.

File: package-rfcx/rfcx/client.py [hunk 4]
```diff
@@ -264,3 +235,25 @@ def streams(self, keyword=None, limit=1000, offset=0):
...
+    def annotations(self, start=None, end=None, classifications=None, stream=None, limit=1000, offset=0):
+        """Retrieve a list of annotations
+
+        Args:
+            start: Minimum timestamp of the audio. If None then defaults to exactly 30 days ago.
+            end: Maximum timestamp of the audio. If None then defaults to now.
+            classifications: (optional, default= None) List of classification names.
+            stream: (optional,default= None) Limit results to a given stream.
+            limit: (optional, default= 1000) Maximum number of results to be return.
+            offset: (optional, default= 0) Number of results to skip.
```
Comment (antonyharfield) [inline package-rfcx/rfcx/client.py:248]: Randoms spaces and lack of space.

```
(optional, default=None)
```

**Review by antonyharfield — APPROVED** [review]
Minor style improvement.

**Pull request merged**